streamingccl: an easy way to set up cluster for tenant streaming in the demo

gh-casper opened the pull request:
This makes starting tenant sql server easier and avoid the need to set up authentication certificates.

Start source cluster:
`make test PKG=./pkg/ccl/streamingccl/streamingest TESTS=TestDemoCluster TESTFLAGS='-v -tenant=10 -num_nodes=3' | grep 'c2c'`

Start destination cluster:
`make test PKG=./pkg/ccl/streamingccl/streamingest TESTS=TestDemoCluster TESTFLAGS='-v -tenant=20 -num_nodes=2' | grep 'c2c'`

Commands to start system SQL and tenant SQL will pop up in the end. Also, there are SQL statements to create a replication stream with initial scan or with resuming from previous checkpoint and complete stream ingestion job.

Stream replication with initial scan: 
`
SET enable_experimental_stream_replication = true; 
RESTORE TENANT {src_tenant_id} FROM REPLICATION STREAM FROM {pg_url} AS TENANT {dest_tenant_id};
`

Stream replication resuming from previous timestamp (previous timestamp can be retrieved by `SELECT cluster_logical_timestamp()`):  
`
SET enable_experimental_stream_replication = true; 
RESTORE TENANT {src_tenant_id} FROM REPLICATION STREAM FROM {pg_url} AS OF SYSTEM TIME {previous_timestamp} AS TENANT {dest_tenant_id};
`

Cut over the replication stream (cutover timestamp can be retrieved by `SELECT clock_timestamp()`):
`SELECT crdb_internal.complete_stream_ingestion_job({ingestion_job_id}, {cutover_timestamp});`

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/75542)
<!-- Reviewable:end -->

gh-casper committed:
streamingccl: an easy way to set up cluster for tenant streaming in the demo

make test PKG=./pkg/ccl/streamingccl/streamingest TESTS=TestDemoCluster TESTFLAGS='-v -src_tenant=10 -num_nodes=3' | grep 'c2c'

make test PKG=./pkg/ccl/streamingccl/streamingest TESTS=TestDemoCluster TESTFLAGS='-v -start_dest_cluster -dest_tenant=20 -num_nodes=2' | grep 'c2c'

Release note: None

shermanCRL commented:
Do we intend to move forward on this, or consider it obviated by:
- https://github.com/cockroachdb/cockroach/issues/83426
